portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -52,15 +60,22 @@ public void init() {
   }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    int maxComments = Integer.parseInt(request.getParameter("max_comments"));
+    int commentsPerPage = Integer.parseInt(request.getParameter("comments_per_page")), pgNumber = Integer.parseInt(request.getParameter("pg_number")) - 1;
```

Comment:
Having these two variables declared on the same line is kind of confusing and (I think) not really standard practice. You should probably put each on its own line.